### ISSUE TITLE
Path tracing broken on Linux

### ISSUE THREAD

**Issue by JolifantoBambla** [allen-cell-animated/agave#38]
## Description

The path tracing fragment shader does not compile on Linux.
There are multiple locations where sampler arrays are indexed using integer variables.
Compilation fails with this error:
```
0000008 | 2022-04-07, 13:46:57.500381 [warning] - GLShader::compile(Fragment): 0:295(22): error: sampler arrays indexed with non-constant expressions are forbidden in GLSL 1.30 and later
```

Here's the complete [logfile.log](https://github.com/allen-cell-animated/agave/files/8445489/logfile.log).

## Reproduction

Start `agave`.

## Environment

OS: Ubuntu 20.04
OpenGL renderer string: AMD RENOIR (LLVM 13.0.0, DRM 3.41, 5.13.0-10039-tuxedo)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.0.1 - kisak-mesa PPA

**Comment by toloudis** [comment]
I don't believe this is Linux, but related to the AMD driver on Linux.   I'll do some research and see what I can do.  Nearly all of my testing has been with nvidia gpus.

**Comment by toloudis** [comment]
Noting possible workaround to try later:  https://github.com/gdkchan/SPICA/issues/47#issuecomment-399712878

**Comment by toloudis** [comment]
possible better fix: change the `version 330 core` to `version 400 core` or higher,  in the shader source code.... for all shaders.

**Comment by toloudis** [comment]
see branch https://github.com/allen-cell-animated/agave/tree/fix/linux-amd-shader-compilation-error

**Comment by JolifantoBambla** [comment]
That did it! Thanks!

**Comment by toloudis** [comment]
Thanks for testing it out!  I'll make a pull request and get it into a release in the next few days.
